src/test/ui/impl-trait/cross-return-site-inference.stderr

```diff
@@ -7,7 +7,9 @@ LL |     Err("whoops")?;
    |                  ^ the trait `From<&str>` is not implemented for `impl Debug`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
-   = help: the trait `FromResidual<Result<Infallible, E>>` is implemented for `Result<T, F>`
+   = help: the following other types implement trait `FromResidual<R>`:
+             <Result<T, F> as FromResidual<Result<Infallible, E>>>
+             <Result<T, F> as FromResidual<Yeet<E>>>
```

Comment:
Technically adding a second impl when there was only one before is a breaking change I think. There are some cases where inference figures something out. Considering that users can't mention FromResidual themselves, we're good here, but probably good to keep in mind.

Reply:
Crater says we're clean, but I think the other thing that helps is the direction in which we're using it -- `AsRef<T>` has this problem all the time when used as a method, because with `self` known then if there's only one impl it's easy to know the `T`.  But here we're going the other direction, and I know know that the inference engine allows taking advantage of "`FromResidual<Concrete>` is only implemented for one type", since that's fundamentally less stable, as anyone who wants *could* implement that themselves later if they want.

(And if there is a way that adding another impl here can be a breaking change, all the better to get a second one in early to keep anyone from finding a way to write stable code that depends on it 🙃)